### ISSUE TITLE
Expandable items may exceed volume capacity when activated within a container.

### ISSUE THREAD

**Issue by Scrivener07** [CleverRaven/Cataclysm-DDA#40775]
# Describe the bug
When an expandable item is inside a container and is activated, the item's volume will increase and possibly exceed the containers capacity.

For example batons do not like to be expanded while still inside your pants. I have found that another "expandable" item, a rolled up sleeping bag will also cause this issue.  This might also apply to other expandable items.

```
 DEBUG    : volume exceeds capacity (1540ml > 1200ml)
 FUNCTION : float item_contents::relative_encumbrance() const
 FILE     : src/item_contents.cpp
 LINE     : 811
src/item_contents.cpp:811 [float item_contents::relative_encumbrance() const] volume exceeds capacity (1540ml > 1200ml)
```

# Steps To Reproduce
1. Start a new game and add an expandable baton or sleeping bag to an inventory pocket.
2. Check that the expanded item volume will exceed the pocket size (baton in pants). Expand the item while its inside the pocket.
3. There will be a volume exceeded exception.

# Expected behavior
A possible solution is to only allow an item to be expanded when its outside a container. With the inventory pocket system, wielding an item before activating it is the only safe option I know of since you cannot activate an item unless its wielded or in one of your pockets. It makes sense that you would have to wield an item to expand it but maybe some thought could be put into how to streamline that micro a bit. Another idea is to just drop the item if its expanded size would exceed the pocket capacity.

# Screenshots
![Error](https://user-images.githubusercontent.com/1912967/82731577-02ac4480-9cd6-11ea-82e1-1a91b5712e1a.PNG)

# Versions and configuration
 - OS: `Windows 7`
 - Game Version: `0.E-2203-g30952d6`
 - Graphics version: `Tiles (dead people)`
 - Ingame language: `English`
 - Mods loaded: [`Disable NPC Needs`]

# Remarks
I no longer have the game save or logs for the game session that this occurred on but if there is any trouble reproducing the issue then I can get some new ones. Its pretty straight forward to reproduce though.

**Comment by starytroll** [comment]
Encountered the same issue, when activating emergency blanket stored in one of my pockets.

**Comment by AnonimusC** [comment]
A condom tried to fill up more than it could hold (water). Volume exceeds capacity.
>  DEBUG    : volume exceeds capacity (1396ml > 1000ml)
>  FUNCTION : float item_contents::relative_encumbrance() const
>  FILE     : src/item_contents.cpp
>  LINE     : 1478
1. Load the [save](https://github.com/CleverRaven/Cataclysm-DDA/files/5608543/save.zip
).
2. Try to collect water in the condom to your right, from the lake.

[debug.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/5608454/debug.zip)
Connected with:
https://github.com/CleverRaven/Cataclysm-DDA/issues/44615

**Comment by l29ah** [comment]
This doesn't crash for me.

**Comment by kevingranade** [comment]
Fixed by #45873
